api/handlers.go

```diff
@@ -239,35 +239,62 @@ func (a *API) Mount(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Only perform the TTL refresh if the driver is in unlocked mode.
-	if !volConfig.Unlocked {
-		if err := a.startTTLRefresh(volName); err != nil {
-			a.HTTPError(w, err)
+	mc, err := driver.Mount(driverOpts)
+	if err != nil {
+		if err := a.Lock.ClearLock(ut, (*a.Global).Timeout); err != nil {
+			a.HTTPError(w, errors.RefreshMount.Combine(errored.New(volConfig.String())).Combine(err))
 			return
 		}
-	}
 
-	mc, err := driver.Mount(driverOpts)
-	if err != nil {
 		a.HTTPError(w, errors.MountFailed.Combine(err))
 		return
 	}
 
 	a.MountCollection.Add(mc)
 
+	// Only perform the TTL refresh if the driver is in unlocked mode.
+	if !volConfig.Unlocked {
+		if err := a.startTTLRefresh(volName); err != nil {
+			a.RemoveStopChan(volName)
+
+			if err := a.clearLock(ut); err != nil {
+				a.HTTPError(w, err)
+				return
+			}
```

Comment:
@erikh I don't know what's the right action to be performed here. but, this change will make the mount at L242 untracked. I suspect it will create issues while snapshotting locked volumes which purely relies on mount locks.